addons/knowledge/models/res_partner.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class Partner(models.Model):
+    _name = 'res.partner'
+    _inherit = 'res.partner'
+
+    def unlink(self):
+        """ This override will delete all the private articles linked to the deleted partners. """
+        self.env['knowledge.article.member'].sudo().search([
+            ('partner_id', 'in', self.ids)]
+        ).article_id.filtered(lambda a: a.category == 'private').unlink()
```

Comment:
Seems like you store the category... so it could be put directly in the search domain right?